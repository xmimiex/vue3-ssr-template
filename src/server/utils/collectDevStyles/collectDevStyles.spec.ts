import { collectCss, hashCode, componentsModules } from '.'


describe('hashCode', () => {

  it('should generate content based hash', () => {
    const content = 'content'
    const contentb = 'contentb'

    expect(hashCode(content)).not.toBe(hashCode(contentb))
  })
})


describe('collectCss', () => {

  it('should collect css', () => {
    const mods = new Set([{
      id: 'mod1.scss',
      ssrModule: {
        default: 'content1',
      },
      importedModules: new Set(),
    },
    {
      id: 'mod2.scss',
      ssrModule: {
        default: 'content2',
      },
      importedModules: new Set([{
        id: 'mod3.scss',
        ssrModule: {
          default: 'content3',
        },
        importedModules: new Set(),
      }]),
    },
    {
      id: 'mod4.js',
      ssrModule: {
        default: 'content4',
      },
      importedModules: new Set(),
    }]) as any

    expect(collectCss(mods)).toMatchSnapshot()
  })
})

describe('componentsModules', () => {

  it('should collect componentsModules', () => {
    const getModulesByFileMock = jest.fn(() => new Set(['mod4.scss', 'mod5.scss', 'mod6.scss']))
    const ViteDevServer = {
      moduleGraph: {
        getModulesByFile: getModulesByFileMock,
      },
    } as any
    const components = ['mod1.scss', 'mod2.scss', 'mod3.scss']
    const mods = componentsModules(components, ViteDevServer)
    expect(getModulesByFileMock).toHaveBeenCalled()
    expect(mods).toEqual(new Set(['mod4.scss', 'mod5.scss', 'mod6.scss']))
  })
})
