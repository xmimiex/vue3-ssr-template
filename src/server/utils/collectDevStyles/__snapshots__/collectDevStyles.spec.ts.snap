// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collectCss should collect css 1`] = `""`;
